tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/SetInclusionRuleWithFunArrays.scala

```diff
@@ -0,0 +1,62 @@
+package at.forsyte.apalache.tla.bmcmt.rules
+
+import at.forsyte.apalache.tla.bmcmt._
+import at.forsyte.apalache.tla.bmcmt.types.{CellTFrom, PowSetT}
+import at.forsyte.apalache.tla.lir.{OperEx, SetT1}
+import at.forsyte.apalache.tla.lir.convenience._
+import at.forsyte.apalache.tla.lir.UntypedPredefs._
+import at.forsyte.apalache.tla.lir.oper.TlaSetOper
+
+class SetInclusionRuleWithFunArrays(rewriter: SymbStateRewriter) extends RewritingRule {
+
+  override def isApplicable(state: SymbState): Boolean = {
+    state.ex match {
+      case OperEx(TlaSetOper.subseteq, _, _) =>
+        true
+
+      case _ =>
+        false
+    }
+  }
+
+  override def apply(state: SymbState): SymbState = {
+    state.ex match {
+      case OperEx(TlaSetOper.subseteq, left, right) =>
+        // switch to cell theory
+        val leftState = rewriter.rewriteUntilDone(state.setRex(left))
+        val leftCell = leftState.arena.findCellByNameEx(leftState.ex)
+        val rightState = rewriter.rewriteUntilDone(leftState.setRex(right))
+        val rightCell = rightState.arena.findCellByNameEx(rightState.ex)
+        (leftCell.cellType, rightCell.cellType) match {
+          case (CellTFrom(SetT1(_)), CellTFrom(SetT1(_))) =>
+            rewriter.lazyEq.subsetEq(rightState, leftCell, rightCell)
+
+          case (CellTFrom(SetT1(SetT1(t1))), PowSetT(SetT1(t2))) =>
+            if (t1 != t2) {
+              throw new RewriterException("Unexpected set types: %s and %s in %s"
+                    .format(t1, t2, state.ex), state.ex)
+            } else {
+              subsetInPowset(rightState, leftCell, rightCell)
+            }
+
+          case _ =>
+            throw new RewriterException("Unexpected set types: %s and %s in %s"
+                  .format(leftCell.cellType, rightCell.cellType, state.ex), state.ex)
+        }
```

Comment:
This isn't exhaustive:
`(PowSetT(SetT1(t2)), CellTFrom(SetT1(SetT1(t1))))` where `t1 = t2` is technically also valid, as are combinations of `FunSetT` and `SetT(FunT)`

Reply:
I will change it to `Not implemented (open an issue)`. If these edge cases become relevant we can add them later. Thanks for spotting this.